doc/compiler/optimizer/InliningAndInliningGuards.md

```diff
@@ -0,0 +1,406 @@
+<!--
+Copyright IBM Corp. and others 2023
+
+This program and the accompanying materials are made available under
+the terms of the Eclipse Public License 2.0 which accompanies this
+distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+or the Apache License, Version 2.0 which accompanies this distribution and
+is available at https://www.apache.org/licenses/LICENSE-2.0.
+
+This Source Code may also be made available under the following
+Secondary Licenses when the conditions for such availability set
+forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+General Public License, version 2 with the GNU Classpath
+Exception [1] and GNU General Public License, version 2 with the
+OpenJDK Assembly Exception [2].
+
+[1] https://www.gnu.org/software/classpath/license.html
+[2] https://openjdk.org/legal/assembly-exception.html
+
+SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+-->
+
+# High Level Outline
+
+1. The Benefits Of Inlining
+2. The Downside Of Inlining
+3. Balance Multiple Paths Of Optimizations
+3. Overview of Guards
+4. Different Types Of Guards
+
+# 1. The Benefits of Inlining
+
+Inliner, local CSE (Common Subexpression Elimination), GRA (Global Register Allocation), and Global Value Propagation
+are some of the most impactful optimizations from the perspective of improving performance throughput.
+These highly effective optimizations are important for scorching and hot compilations as well as for a flat
+profile with warm compilations.
+
+## Basic Concepts
+- **Call site** ([TR_CallSite](https://github.com/eclipse/omr/blob/88427d6420f5fa1bf00f897d1f589390bf2cafae/compiler/optimizer/CallInfo.hpp#L268))
+is the code that makes the call. It contains the call node (caller) that makes the call and the call target
+(callee) that is inlined into the call site.
+- **Call target** ([TR_CallTarget](https://github.com/eclipse/omr/blob/88427d6420f5fa1bf00f897d1f589390bf2cafae/compiler/optimizer/CallInfo.hpp#L130))
+represents the callee or the specific method that is inlined into the call site.
+
+## Eliminate Calling Overhead
+
+The benefits of inlining are wide ranging. An application with a flat profile generally has a lot of compiled
+methods. The calling overhead in a flat profile could add up across all methods. The most direct benefit of
+inlining is to get rid of call overhead such as call instruction, return instruction, prologue, epilogue, etc.
+**The key question when it comes to inlining is how much is going to be executed versus how much code there is
```

Comment:
Maybe you meant "how much the call overhead is versus how much code there is in the callee method"